src/main/java/org/zaproxy/zap/extension/hud/HudApiProxy.java

```diff
@@ -64,11 +68,26 @@ public void addCustomHeaders(String name, RequestType type, HttpMessage msg) {
 
     @Override
     public String handleCallBack(HttpMessage msg) throws ApiException {
+        String url = msg.getRequestHeader().getURI().toString();
         try {
 
-            String url = msg.getRequestHeader().getURI().toString();
             LOG.debug("API proxy callback url = " + url);
 
+            if (!api.getZapHudCookieValue()
+                    .equals(api.getRequestCookieValue(msg, HudAPI.ZAP_HUD_COOKIE))) {
+                String errorMsg =
+                        "Correct cookie not supplied when accessing API "
```

Comment:
The negative logic in this error message is difficult. Perhaps "Incorrect cookie supplied when accessing the API "

Reply:
+1